index/scorch/snapshot_segment.go

```diff
@@ -35,6 +35,12 @@ type SegmentSnapshot struct {
 	deleted *roaring.Bitmap
 	creator string
 
+	// this flag is needed to identify whether this
+	// segment was mmaped recently, in which case
+	// we consider the loading cost of the metadata
+	// as part of IO stats.
+	mmaped uint64
```

Comment:
Move this to be the first attribute of this struct. Also, a uint32 would suffice?

Reply:
yeah it's sufficient, will do.